AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/InitialSync/SyncEventEmitterTests.swift

```diff
@@ -0,0 +1,192 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSPluginsCore
+@testable import AWSDataStoreCategoryPlugin
+
+class SyncEventEmitterTests: XCTestCase {
+    var initialSyncOrchestrator: MockAWSInitialSyncOrchestrator?
+    var reconciliationQueue: MockAWSIncomingEventReconciliationQueue?
+    var syncEventEmitter: SyncEventEmitter?
+    var expectedModelSyncedEventPayloads: [ModelSyncedEvent]!
+
+    override func setUp() {
+        super.setUp()
+
+        ModelRegistry.reset()
+        MockModelReconciliationQueue.reset()
+        MockAWSInitialSyncOrchestrator.reset()
+
+        expectedModelSyncedEventPayloads = [ModelSyncedEvent(modelName: "Post",
+                                                             isFullSync: true, isDeltaSync: false,
+                                                             added: 1, updated: 0, deleted: 0),
+                                            ModelSyncedEvent(modelName: "Comment",
+                                                             isFullSync: true, isDeltaSync: false,
+                                                             added: 1, updated: 0, deleted: 0)]
+    }
+
+    /// - Given: A SyncEventEmitter
+    /// - When:
+    ///    - One model is registered
+    ///    - Perform an initial sync
+    /// - Then:
+    ///    - One modelSynced event should be received
+    ///    - One syncQueriesReady event should be received
+    func testModelSyncedAndSyncQueriesReadyWithOneModelRegistered() throws {
+        let modelSyncedReceived = expectation(description: "modelSynced received")
+        let syncQueriesReadyReceived = expectation(description: "syncQueriesReady received")
+
+        ModelRegistry.register(modelType: Post.self)
+        let testPost = Post(id: "1", title: "post1", content: "content", createdAt: .now())
+        let anyPost = AnyModel(testPost)
+        let anyPostMetadata = MutationSyncMetadata(id: "1",
+                                               deleted: false,
+                                               lastChangedAt: Int(Date().timeIntervalSince1970),
+                                               version: 1)
+        let anyPostMutationSync = MutationSync<AnyModel>(model: anyPost, syncMetadata: anyPostMetadata)
+        let postMutationEvent = try MutationEvent(untypedModel: testPost, mutationType: .create)
```

Comment:
suggestion: there is going to be some commonalities across your tests in this class that a method like `getMutationEvent(post, .create)` may be helpful to reduce the duplication of code

Reply:
I realized that a func getMutationEvent(post, .create) looks like:
```swift
private func getMutationEvent(anyModel: AnyModel, mutationType: MutationEvent.MutationType) throws -> MutationEvent {
    return try MutationEvent(untypedModel: anyModel, mutationType: .create)
}
```
Seems that not many sense to put it as a separate function?

And I also tried reduce the duplication of code. Like moved
```swift
let testPost = Post(id: "1", title: "post1", content: "content", createdAt: .now())
        let anyPost = AnyModel(testPost)
        let anyPostMetadata = MutationSyncMetadata(id: "1",
                                                   deleted: false,
                                                   lastChangedAt: Int(Date().timeIntervalSince1970),
                                                   version: 1)
        let anyPostMutationSync = MutationSync<AnyModel>(model: anyPost, syncMetadata: anyPostMetadata)

        let postMutationEvent = try MutationEvent(untypedModel: testPost, mutationType: .create)
```
into `setUp()`, but it fails the test.

Reply:
looks like you have increased code coverage with more variations of inputs so there's not much you can factor out, without it becoming a method with lots parameters.. resolving this comment